src/get.rs

```diff
@@ -45,13 +45,18 @@ impl Default for Options {
     }
 }
 
-/// Setup a QUIC connection to the provided address.
-async fn setup(opts: Options) -> Result<quinn::Connection> {
+/// Create a quinn client endpoint
+pub fn make_client_endpoint(
+    peer_id: Option<PeerId>,
+    alpn_protocols: Vec<Vec<u8>>,
```

Comment:
could be `&[&'static u8]]` probably

Reply:
You would have to turn it into a Vec<Vec<u8>> to actually set it because rustls::ClientConfig takes a Vec<Vec<u8>>